lib/webpacker/node_bundler.rb

```diff
@@ -0,0 +1,67 @@
+# Finds available global or local node package manager - yarn or npm
+
+require "webpacker/configuration"
+
+module Webpacker::NodeBundler
+  extend self
+
+  def name
+    @package ||= packages.find(-> { raise Errno::ENOENT }) { |package| available?(package) }
+
+  rescue Errno::ENOENT
+    $stderr.puts "Node package managers npm and yarn not found! Scanned following paths for executables - #{packages}"
+    $stderr.puts "Make sure either npm or yarn is installed globally, or locally within your app node_modules folder."
+    exit!
+  end
+
+  def command
+    commands[name]
+  end
+
+  def command_dev
+    commands[:dev][name]
+  end
+
+  private
+
+    def available?(package)
+      null_device = Gem.win_platform? ? "/nul 2>&1" : "/dev/null 2>&1"
+      system("#{package} --version > #{null_device}")
```

Comment:
Many developers will have both yarn and npm installed. 

Seems like we just pick yarn if first. Should that be documented? Maybe this should just be in the config file or via an ENV, b/c those devs working on multiple projects will surely have yarn.